### PR TITLE
[Scrutinizer] Fix unused code issues in boot and include

### PR DESCRIPTION
**Pull request by MrPetovan**
I finally set up Scrutinizer thanks to @Rudloof from whom I stole the configuration file, and the results are impressive, about 1800 PHP issues that we should fix at some point.

https://scrutinizer-ci.com/g/MrPetovan/friendica/issues/task/fix-scrutinizer-issues

This PR is only about unused code issues and formatting in `boot.php` and the `include` folder.

I'll progress slowly towards fixing all issues and trying to keep PR size manageable.

### CODE REVIEW

**Comment by annando** [comment]
I have the feeling that this PR will cause massive trouble with https://github.com/friendica/friendica/pull/4160 ...

**Comment by MrPetovan** [comment]
It's possible, and I'll take care of any conflicts.

File: include/api.php [hunk 1]
```diff
@@ -1116,12 +1111,7 @@ function api_statuses_update($type)
 		throw new ForbiddenException();
 	}
 
-	$user_info = api_get_user($a);
```
Comment (annando) [inline include/api.php]: I have the the feeling as if ```api_get_user``` is also used to check if the user was authenticated. See the code here: https://github.com/friendica/friendica/blob/develop/include/api.php#L588 

AFAIK this code will be executed whenever the function is called like above and the user hadn't authenticated yet.

File: include/datetime.php [hunk 2]
```diff
@@ -180,10 +180,6 @@ function datetime_convert($from = 'UTC', $to = 'UTC', $s = 'now', $fmt = "Y-m-d
 function dob($dob) {
 	list($year,$month,$day) = sscanf($dob,'%4d-%2d-%2d');
 
-	$f = Config::get('system', 'birthday_input_format');
```
Comment (annando) [inline include/datetime.php]: Please remove the documentation for this config value from "doc/htconfig.md" as well.

File: mod/events.php [hunk 3]
```diff
@@ -475,11 +475,6 @@ function events_content(App $a) {
 		$fhour   = ((x($orig_event)) ? datetime_convert('UTC', $tz, $fdt, 'H') : 0);
 		$fminute = ((x($orig_event)) ? datetime_convert('UTC', $tz, $fdt, 'i') : 0);
 
-		$f = Config::get('system','event_input_format');
```
Comment (annando) [inline mod/events.php]: Like above: Please remove the documentation for the config value as well.

**Comment by MrPetovan** [comment]
Turns out it was much less a problem than expected! 👍

File: boot.php [hunk 4]
```diff
@@ -798,7 +795,6 @@ function run_update_function($x)
 		Config::set('system', 'build', $x + 1);
 		return true;
```
[the remark this answers is not in the document]
Comment (MrPetovan) [reply boot.php]: The one below was never reached.

File: include/api.php (hunk 1 above)
Comment (MrPetovan) [reply include/api.php]: Nice catch, I should just have removed the variable assignment.

**Pull request merged**